std/algorithm/comparison.d

```diff
@@ -892,116 +892,367 @@ nothrow pure @safe unittest
     auto t = task!cmp("foo", "bar");
 }
 
+private static template allSameTypeIterative(V...)
+{
+    static if (V.length >= 2)
+    {
+        static foreach (Vi; V[1 .. $])
+        {
+            static if (!is(typeof(allSameTypeIterative) == bool) && // not yet defined
+                       !is(V[0] == Vi)) // 10% faster than `!isSame(V[0], Vi)`
+            {
+                enum allSameTypeIterative = false;
+            }
+        }
+    }
+    static if (!is(typeof(allSameTypeIterative) == bool)) // if not yet defined
+    {
+        enum allSameTypeIterative = true;
+    }
+}
+
+/** Check if `pred` is a standard equality string or lambda predicate.
```

Comment:
Private symbols shouldn't have a public documentation. Reduce the number of stars from 2 to 1 to make the documentation hidden. Also the DStyle is against leading stars:

https://dlang.org/dstyle.html#phobos_documentation